### ISSUE TITLE
get_velocity_at_local_position is a global offset from the body origin, not a local point

### ISSUE THREAD

**Issue by ghsoares** [godotengine/godot-docs#7110]
**Your Godot version:**
v4.0.1.stable.official.cacf49999

**Issue description:**
In the [PhysicsDirectBodyState2D::get_velocity_at_local_position](https://docs.godotengine.org/en/stable/classes/class_physicsdirectbodystate2d.html#class-physicsdirectbodystate2d-method-get-velocity-at-local-position) and [PhysicsDirectBodyState3D::get_velocity_at_local_position](https://docs.godotengine.org/en/stable/classes/class_physicsdirectbodystate3d.html#class-physicsdirectbodystate3d-method-get-velocity-at-local-position) documentation pages, is described that the input position is a local point in the body transform space, including both translation and rotation, but actually for the math performed (see [this line](https://github.com/godotengine/godot/blob/ba27db36e935f0f5c95df8d57f842d564b7e9b3c/servers/physics_2d/godot_body_2d.h#L321) and [this line](https://github.com/godotengine/godot/blob/ba27db36e935f0f5c95df8d57f842d564b7e9b3c/servers/physics_3d/godot_body_3d.h#L322)) it expects a global offset from the body origin, being the same as `PhysicsDirectBodyState2D::apply_force` for example, it means that the point isn't local in the body rotation, only being relative to the body origin.

I ain't sure if this is the case, but in my project, I was trying the local point and the global offset, while the global offset returned the correct value. If it is really incorrect, I'll be more than happy to contribute to the docs fix.

**URL to the documentation page:**
[PhysicsDirectBodyState2D](https://docs.godotengine.org/en/stable/classes/class_physicsdirectbodystate2d.html) and [PhysicsDirectBodyState3D](https://docs.godotengine.org/en/stable/classes/class_physicsdirectbodystate3d.html)

**Comment by Code-Tamashi** [comment]
@ghsoares : Thanks to you, I was able to finally understand what **get_velocity_at_local_position** expects in input.
I confirm, you are right.

I'm using this method in a raycast car project and after a car rotation, at a certain point, the Vector3 returned by  **get_velocity_at_local_position** is flipped in all axis/direction.

get_velocity_at_local_position (wheel.position) => issue after car rotation.
get_velocity_at_local_position **(wheel.global_position - car.global_position)** => works for me.

I must say that the documentation is **really misleading** as the input parameter should **not** be local at all.
I 'm not sure it's a bug as it works great when you use the correct input parameter with this method.

Hope it can help.

**Comment by novaplusplus** [comment]
I ran into this recently and lost a few years off my life trying to figure it out!

Ideally I think the function name should be changed, but given how far along the engine is, I understand that breaking changes like that are not desirable. I may make a PR with revised docs later once I can remember how to do it, and how best to phrase it.

**Comment by ghsoares** [comment]
I agree that the names should be changed, and it's possible to do (atleast for projects using GDScript), using the project conversion tool to rename references in scripts from the old name to the new name.

Or, if it's viable, make it a separate function with different name, and make `get_velocity_at_local_position` be an alias to the new name, making the function with the old name deprecated for a smoother transition.

**Comment by ghsoares** [comment]
But a more urgent goal would be to update the docs for the function to describe the correct behaviour of the expected space for the parameters.

**Comment by ghsoares** [comment]
Created a PR to address the documentation update: godotengine/godot#116106
